application/src/main/kotlin/net/corda/v5/application/persistence/Query.kt

```diff
@@ -0,0 +1,15 @@
+package net.corda.v5.application.persistence
+
+/**
+ * Used to build a Query.
+ *
+ * @param R the type of the results.
+ */
+interface Query<R> {
+    /**
+     * Execute the [Query]
+     *
+     * @return list of entities found. Empty list if none were found.
+     */
+    fun execute(): List<R>
```

Comment:
Is this the point where we'd go to the DB worker to run it? If so this needs to be marked `@Suspendable`. (We should probably do that for all APIs anyway as a hedge, even if it's not actually going to suspend, as adding the annotation later is not binary compatible with an old version without it.)

Reply:
I added it on `execute()` and left it on the methods in the `persistenceService`. Is that what you meant?

Reply:
Yeah that seems fine. Can't imagine we'd want to suspend setting parameters!